### PR TITLE
Feature/ex 110 use camera and mic hook

### PR DESCRIPTION
**Pull request by kamil-stasiak**

### CODE REVIEW

File: examples/readme.md [hunk 1]
```diff
@@ -11,3 +11,5 @@ List of examples:
   - start broadcasting screen
   - receive media from other peers
   - almost no UI
+
+// todo add description
```
Comment (kamil-stasiak) [inline examples/readme.md:15]: Fix toto

File: examples/use-camera-and-microphone-example/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# React + TypeScript + Vite
```
Comment (kamil-stasiak) [inline examples/use-camera-and-microphone-example/README.md:1]: Fix readme

File: examples/use-camera-and-microphone-example/public/vite.svg
Comment (kamil-stasiak) [inline examples/use-camera-and-microphone-example/public/vite.svg:1]: Remove

File: examples/use-camera-and-microphone-example/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+//  webRTCEndpoint.js:991 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getSenders')
+//     at WebRTCEndpoint2.findSender (webRTCEndpoint.js:991:32
+// {
+//     streamWhenConnected: false,
+//     startStreamingWhenDeviceReady: true,
+//     startOnMount: false,
+//     storage: true,
+// }
+// Click connect -> start devices
+// Jeżeli zaczniesz streamować urządzenia bardzo szybko to wybucha.
```
Comment (kamil-stasiak) [inline examples/use-camera-and-microphone-example/src/App.tsx:24]: Report an error

File: examples/use-camera-and-microphone-example/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    camera: {
+      trackConstraints: VIDEO_TRACK_CONSTRAINTS,
+      autoStreaming: videoAutoStreaming,
+      preview: videoPreview,
+      defaultTrackMetadata: DEFAULT_VIDEO_TRACK_METADATA,
+      // defaultSimulcastConfig: {
+      //   enabled: true,
+      //   active_encodings: ["l", "m", "h"]
+      // }
+    },
```
Comment (kamil-stasiak) [inline examples/use-camera-and-microphone-example/src/App.tsx:76]: uncomment

File: src/create.tsx [hunk 5]
```diff
@@ -259,6 +287,10 @@ const onConnect = <PeerMetadata, TrackMetadata>(
     action.dispatch({ type: "onDisconnected" });
   });
 
+  // important jeżeli użytkownik zapnie się na eventy z JellyfishClient
+  //  to one mogą się odpalić wcześniej niż te tutaj zasetupowane metody
+  //  tak się dzieje na przykład z eventem joined gdy chcę od razu zacząć
+  //  przesyłać tracki zaraz po dołączeniu
```
Comment (kamil-stasiak) [inline src/create.tsx:293]: Report an error and remove this coment

**Comment by Crackhoff** [comment]
When I connect and stop streaming either audio or video sometimes whole room crashes

**Review by Crackhoff — CHANGES_REQUESTED** [review]
I love the hook itself, but I have some questions about the example

File: examples/use-camera-and-microphone-example/src/AudioVisualizer.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    }
+
+    renderFrame();
+  }, [stream]);
+
+
+  return (
+    <div className="flex flex-row flex-nowrap justify-center border-4">
+      <canvas ref={canvasRef} width={200} height={100}/>
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply examples/use-camera-and-microphone-example/src/AudioVisualizer.tsx]: Fixed. Now the canvas width matches the width of its parent container

File: examples/use-camera-and-microphone-example/src/App.tsx [hunk 7]
```diff
@@ -0,0 +1,327 @@
+import "./App.css";
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply examples/use-camera-and-microphone-example/src/App.tsx]: Removed

File: examples/use-camera-and-microphone-example/src/App.css
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply examples/use-camera-and-microphone-example/src/App.css:1]: Removed

File: examples/use-camera-and-microphone-example/src/App.tsx [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+
+                <VideoPlayer stream={video?.stream} />
+              }
+              {audio?.track?.kind === "audio" &&
+                <AudioVisualizer stream={audio?.stream} />
+              }
+            </div>
+          </div>
+          <div>
+            Streaming
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply examples/use-camera-and-microphone-example/src/App.tsx]: Fixed. Now it looks like this:

**Comment by kamil-stasiak** [comment]
> Bug: When I connect and stop streaming either audio or video sometimes whole room crashes

I think it's a bug in Jellyfish, and I believe it's already fixed. Please check the latest commit on the 'main' branch

> When I have auto streaming on and I connect to the server, on dashboard I see two video tracks. What's the cause? If one sends both audio and video, why there is another? If they're both video-only, what happens with audio and why there are two of them?

And I think it'a a bug in `jellyfish-dashboard`. We need to address it in a separate task

File: readme.md [hunk 9]
```diff
@@ -19,93 +20,117 @@ It was tested with `nodejs` version mentioned in `.tool-versions` file.
...
-For pure TypeScript usage, see [Jellyfish TS client](https://github.com/jellyfish-dev/react-client-sdk/tree/main/src/jellyfish).
+For pure TypeScript usage,
+see [Jellyfish TS client](https://github.com/jellyfish-dev/react-client-sdk/tree/main/src/jellyfish).
 
 Prerequisites:
 
 - Running [Jellyfish](https://github.com/jellyfish-dev/jellyfish) server.
 - Created room and token of peer in that room.
-  You u can use [dashboard](https://github.com/jellyfish-dev/react-client-sdk/tree/main/examples/dashboard) example to create room and peer token.
+  You u can use [dashboard](https://github.com/jellyfish-dev/jellyfish-dashboard) to create room and peer token.
```
Comment (Crackhoff) [inline readme.md:30]: typo, "u"

**Review by Crackhoff — APPROVED** [review]
All good!

**Pull request merged**